api/login/login.go

```diff
@@ -2,19 +2,60 @@ package login
 
 import (
 	"api/database"
+	"os"
+	"time"
 
 	"encoding/json"
 	"fmt"
-	"github.com/gorilla/mux"
 	"log"
 	"net/http"
+
+	"github.com/golang-jwt/jwt/v5"
+	"github.com/gorilla/mux"
 )
 
 type Account struct {
 	Email    string `json:"email"`
 	Password string `json:"password"`
 }
 
+// Get the user a JWT token is for, or an error if the token is invalid
+func ValidateToken(tokenString string) (string, error) {
+	token, err := jwt.ParseWithClaims(tokenString, &jwt.RegisteredClaims{}, func(token *jwt.Token) (interface{}, error) {
+		return []byte(os.Getenv("JWT_SECRET")), nil
+	})
+
+	if err != nil {
+		return "", err
+	}
+
+	claims, ok := token.Claims.(*jwt.RegisteredClaims)
+	if !ok || !token.Valid || time.Now().After(claims.ExpiresAt.Time) {
+		return "", fmt.Errorf("invalid token")
+	}
+
+	return claims.Subject, nil
+}
+
+// Get a JWT token for a user, or an error if the user is invalid
+func GetToken(username string, password string) (*jwt.Token, error) {
+	exists := false
+	database.QueryValue(&exists, "SELECT SHA2(?, 256)=password FROM Accounts WHERE email=?", password, username)
+
+	if !exists {
+		return nil, fmt.Errorf("invalid username or password")
+	}
+
+	token := jwt.NewWithClaims(jwt.SigningMethodHS256, jwt.RegisteredClaims{
+		IssuedAt:  jwt.NewNumericDate(time.Now()),
+		ExpiresAt: jwt.NewNumericDate(time.Now().Add(time.Hour * 24)),
+		Issuer:    "LunchBuddies",
+		Subject:   username,
```

Comment:
Follow through previous comment here, also make the key "Email" to be more concise so we know what is stored

Reply:
Do you mean replace `Subject` with `Email`? I chose subject because that is the standardized claim for this in RFC7519 (https://www.rfc-editor.org/rfc/rfc7519#page-9), though I could still make the change if preferred since the standard indicates that it's optional

Reply:
Okay that's all right then, you can leave it as subject